src/components/TodoList.tsx

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable import/no-cycle */
+
+import { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { Filter } from '../types/enum';
+import { TodoContext } from '../TodoContext';
+import { TodoContextProps } from '../types/interfaces';
+
+export const TodoList = () => {
+  const { state, tempTodo } = useContext(TodoContext) as TodoContextProps;
+
+  const filteredTodos = state.todos.filter(todo => {
+    switch (state.filter) {
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      case Filter.All:
+        return true;
+
+      default:
+        return true;
+    }
+  });
```

Comment:
Moving it to `services` would be fine, and I don't know if you really need `case Filter.All:` cuz you get it by default.